### PR TITLE
Sr nits and clars

### PR DESCRIPTION
**Pull request by marco-tiloca-sics**

### CODE REVIEW

File: draft-ietf-core-oscore-edhoc.md [hunk 1]
```diff
@@ -210,7 +210,11 @@ EDHOC verification                                           |
 
 To this end, the specific approach defined in this section consists of sending a single EDHOC + OSCORE request, which conveys the pair (C_R, EDHOC message_3) within an OSCORE-protected CoAP message.
 
-That is, the EDHOC + OSCORE request is in practice the OSCORE Request from {{fig-non-combined}}, as still sent to a protected resource and with the correct CoAP method and options intended for accessing that resource. At the same time, the EDHOC + OSCORE request also transports the pair (C_R, EDHOC message_3) required for completing the EDHOC session. Note that, as specified in {{client-processing}}, C_R is transported in the OSCORE Option rather than in the request payload.
+That is, the EDHOC + OSCORE request is composed of following two parts combined together in a single CoAP message.
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]:
```suggestion
That is, the EDHOC + OSCORE request is composed of the following two parts combined together in a single CoAP message:
```

(two changes)

File: draft-ietf-core-oscore-edhoc.md [hunk 2]
```diff
@@ -210,7 +210,11 @@ EDHOC verification                                           |
...
+
+* The OSCORE Request from {{fig-non-combined}}, which is also in this case sent to a protected resource, with the correct CoAP method and options intended for accessing that resource.
+
+* EDHOC data consisting of the pair (C_R, EDHOC message_3) required for completing the EDHOC session. Note that, as specified in {{client-processing}}, C_R is transported in the OSCORE Option of the OSCORE Request rather than in the request payload.
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: Should you mention that the EDHOC data are "prepended" (not just "in") the request payload?

File: draft-ietf-core-oscore-edhoc.md [hunk 3]
```diff
@@ -220,7 +224,7 @@ The rest of this section specifies how to transport the data in the EDHOC + OSCO
 
 This section defines the EDHOC Option. The option is used in a CoAP request, to signal that the request payload conveys both an EDHOC message_3 and OSCORE-protected data, combined together.
 
-The EDHOC Option has the properties summarized in {{fig-edhoc-option}}, which extends Table 4 of {{RFC7252}}. The option is Critical, Safe-to-Forward, and part of the Cache-Key. The option MUST occur at most once and is always empty. If any value is sent, the value is simply ignored. The option is intended only for CoAP requests and is of Class U for OSCORE {{RFC8613}}.
+The EDHOC Option has the properties summarized in {{fig-edhoc-option}}, which extends Table 4 of {{RFC7252}}. The option is Critical, Safe-to-Forward, and part of the Cache-Key. The option MUST occur at most once and MUST be empty. If any value is sent, the recipient MUST ignore it. Future documents may update the definition of the option, by expanding its semantics and specifying admitted values. The option is intended only for CoAP requests and is of Class U for OSCORE {{RFC8613}}.
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: I would parenthesize the "Future..." sentence (it really is a note).

File: draft-ietf-core-oscore-edhoc.md [hunk 4]
```diff
@@ -280,6 +284,8 @@ The client prepares an EDHOC + OSCORE request as follows.
 
 With the same server, the client SHOULD NOT have multiple simultaneous outstanding interactions (see {{Section 4.7 of RFC7252}}) such that: they consist of an EDHOC + OSCORE request; and their EDHOC data pertain to the EDHOC session with the same connection identifier C_R.
 
+An exception applies for clients that operate under particular time constraints over particularly unreliable networks, thus raising the chances to promptly complete the EDHOC execution with the server through multiple, simultaneous EDHOC + OSCORE requests. As discussed in {{security-considerations}}, this does not have any impact in terms of security.
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: This seems like this procedure is a normative part of EDHOC-OSCORE.  Maybe parenthesize the whole paragraph and s/applies/might apply/...  Just to tone it down a bit.

File: draft-ietf-core-oscore-edhoc.md [hunk 5]
```diff
@@ -394,15 +399,15 @@ The Initiator selects an EDHOC Connection Identifier C_I as follows.
 
 The Initiator MUST choose a C_I that is neither used in any current EDHOC session as this peer's EDHOC Connection Identifier, nor the Recipient ID in a current OSCORE Security Context where the ID Context is not present.
 
-The chosen C_I SHOULD NOT be the Recipient ID of any current OSCORE Security Context.
+The chosen C_I SHOULD NOT be the Recipient ID of any current OSCORE Security Context. Unless the two peers concurrently use alternative methods to establish OSCORE Security Contexts, this allows the Responder to always omit the 'kid context' in the OSCORE Option of its messages sent to the Initiator, when protecting those with an OSCORE Security Context where C_I is the Responder's OSCORE Sender ID (see {{Section 6.1 of RFC8613}}).
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: The new text is again a note and not new normative content, so I'd add a "note that".

File: draft-ietf-core-oscore-edhoc.md [hunk 6]
```diff
@@ -394,15 +399,15 @@ The Initiator selects an EDHOC Connection Identifier C_I as follows.
...
 
 ### Responder Processing of Message 2
 
 The Responder selects an EDHOC Connection Identifier C_R as follows.
 
 The Responder MUST choose a C_R that is neither used in any current EDHOC session as this peer's EDHOC Connection Identifier, nor is equal to the EDHOC Connection Identifier C_I specified in the EDHOC message_1 of the present EDHOC session (i.e., after its decoding as per {{Section 3.3 of I-D.ietf-lake-edhoc}}), nor is the Recipient ID in a current OSCORE Security Context where the ID Context is not present.
 
-The chosen C_R SHOULD NOT be the Recipient ID of any current OSCORE Security Context.
+The chosen C_R SHOULD NOT be the Recipient ID of any current OSCORE Security Context. Unless the two peers concurrently use alternative methods to establish OSCORE Security Contexts, this allows the Initiator to always omit the 'kid context' in the OSCORE Option of its messages sent to the Responder, when protecting those with an OSCORE Security Context where C_R is the Initiator's OSCORE Sender ID (see {{Section 6.1 of RFC8613}}).
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: Hmm, can we remove some of the duplication (", for a reason analogous to the one given above with C_I" or some such?)

File: draft-ietf-core-oscore-edhoc.md [hunk 7]
```diff
@@ -448,7 +451,9 @@ The following parameters are defined.
 
 * 'ed-ead', specifying the support of the server for an External Authorization Data (EAD) item (see {{Section 3.8 of I-D.ietf-lake-edhoc}}). This parameter MUST specify a single value, which is taken from the 'Label' column of the "EDHOC External Authorization Data" registry defined in {{Section 9.5 of I-D.ietf-lake-edhoc}}. This parameter MAY occur multiple times, with each occurrence specifying the ead_label of an EAD item that the server supports.
 
-* 'ed-comb-req', specifying, if present, that the server supports the EDHOC + OSCORE request defined in {{edhoc-in-oscore}}. A value MUST NOT be given to this parameter and any present value MUST be ignored by parsers.
+* 'ed-comb-req', specifying, if present, that the server supports the EDHOC + OSCORE request defined in {{edhoc-in-oscore}}. A value MUST NOT be given to this parameter and any present value MUST be ignored by the recipient.
+
+Future documents may update the definition of the parameters 'ed-i', 'ed-r', and 'ed-comb-req', by expanding their semantics and specifying admitted values.
```
Comment (cabo) [inline draft-ietf-core-oscore-edhoc.md]: (Another parenthesis?)

File: draft-ietf-core-oscore-edhoc.md (hunk 1 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Done.

File: draft-ietf-core-oscore-edhoc.md (hunk 2 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Of the EDHOC data, only EDHOC message_3 is prepended to the payload of the OSCORE Request, while C_R is already specified in the OSCORE Option of that OSCORE Request.

The former point was related to the second from last paragraph of Section 3.0, now moved within the bullet point on the EDHOC data that also clarifies the prepending.

File: draft-ietf-core-oscore-edhoc.md (hunk 3 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Done.

File: draft-ietf-core-oscore-edhoc.md (hunk 4 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Done.

File: draft-ietf-core-oscore-edhoc.md (hunk 5 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Done.

File: draft-ietf-core-oscore-edhoc.md (hunk 6 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: I used your suggestion to replace "unless the two peers concurrently use alternative methods to establish OSCORE Security Contexts", while the following words are more mirrored than duplicated.

File: draft-ietf-core-oscore-edhoc.md (hunk 7 above)
Comment (marco-tiloca-sics) [reply draft-ietf-core-oscore-edhoc.md]: Done.

**Review by cabo — APPROVED** [review]
Changes look good; the remaining duplication on C_I/C_R is not necessary to fix.

**Review by cabo — APPROVED** [review]
Thank you!

**Pull request merged**